earth_enterprise/docker/bin/build-docker-image.sh

```diff
@@ -1,33 +1,57 @@
 #! /bin/bash
 
-# Set these variables before running this script to override the
-# default values below:
-: ${STAGE_1_IMAGE_NAME:="opengee-experimental-stage-1"}
-: ${STAGE_2_IMAGE_NAME:="opengee-experimental-stage-2"}
-: ${OUTPUT_IMAGE_NAME:="opengee-experimental"}
+# This script builds a Docker image containing Open GEE Server and Fusion.
+
+source "$(dirname "$0")/lib/input-variables.sh"
+source "$(dirname "$0")/lib/image-building.sh"
+
+
+# Look at `build_image_parse_input_variables` in <lib/input-variables.sh> for
+# input variables you can set before running this script.
+
+build_image_parse_input_variables
+
+: ${OUTPUT_IMAGE_NAME:="opengee-experimental-${OS_DISTRIBUTION}${CLONE_SUFFIX}"}
 : ${STAGE_1_CONTAINER_NAME:="$STAGE_1_IMAGE_NAME-build-$(date '+%s.%N')-$RANDOM"}
 : ${STAGE_2_CONTAINER_NAME:="$STAGE_2_IMAGE_NAME-build-$(date '+%s.%N')-$RANDOM"}
-
-STAGE_COMPLETE_MESSAGE="Open GEE Docker Stage 2: Entering wait loop."
+: ${FLATTEN_IMAGE:="true"}
+# Set TST_DOCKER_QUIET to a non-empty string to avoid prompting the user for input.
 
 SELF_NAME=$(basename "$0")
 SELF_DIR=$(dirname "$0")
+DOCKER_DIR="$SELF_DIR/.."
 
-cd "$SELF_DIR/../image-definition" || exit 1
-docker build -f Dockerfile.stage-1 --no-cache=true --rm=true -t "${STAGE_1_IMAGE_NAME}" .
-docker run --cap-add=DAC_READ_SEARCH --name="$STAGE_1_CONTAINER_NAME" -ti "${STAGE_1_IMAGE_NAME}" |
-while read ln; do
-    echo "$ln"
-    # Wait for install to complete:
-    case "$(echo "$ln" | tr --delete '\r\n')" in
-        "$STAGE_COMPLETE_MESSAGE")
-            echo "$SELF_NAME: Flattening Docker image."
-            docker export "$STAGE_1_CONTAINER_NAME" | docker import - "$STAGE_2_IMAGE_NAME"
-            STAGE_2_CONTAINER_ID=$(docker run -d "$STAGE_2_IMAGE_NAME" /root/opengee/bin/spin.sh)
-            docker commit --change="CMD /root/opengee/bin/start-servers.sh" "$STAGE_2_CONTAINER_ID" "$OUTPUT_IMAGE_NAME"
-            docker stop "$STAGE_2_CONTAINER_ID"
+function is_string_false()
+{
+    case $(echo "$1" | tr '[:upper:]' '[:lower:]') in
+        false|no|0)
```

Comment:
It's more of a default to `True`, unless the value is explicitly `False` function.

It could treat an empty string as `False`, though the way it is used in this script, the tested value couldn't be empty, since it would default to `true` if it was empty.